packages/playwright-core/src/server/frames.ts

```diff
@@ -1345,11 +1345,10 @@ export class Frame extends SdkObject {
       omitAttached = true;
     else if (options.isNot && options.expression === 'to.have.count' && options.expectedNumber !== 0)
       omitAttached = true;
-    else if (!options.isNot && options.expression.endsWith('.array') && options.expectedText!.length === 0)
+    else if (!options.isNot && options.expression.endsWith('.array') && options.expectedText?.length === 0)
```

Comment:
I think this does not work with `contain.class.array` because `undefined !== 0`, but it should.